### PR TITLE
Using commons-lang3 to replace commons-lang

### PR DESCRIPTION
**Pull request by ruanwenjun**
## What's the purpose of this PR

1. It's better to use `commons-lang3` rather than `commons-lang`, since `commons-lang` is deprecated.
2. It's better to use `org.apache.commons.lang3.SystemUtils` to judge the system, this utils is more reliable.

## Which issue(s) this PR fixes:
no

## Brief changelog

* Using `commons-lang3` to replace `commons-lang`
* Using `org.apache.commons.lang3.SystemUtils` to judge is Window.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [ ] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [ ] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by ruanwenjun** [comment]
I have read the CLA Document and I hereby sign the CLA

File: apollo-core/src/main/java/com/ctrip/framework/foundation/internals/Utils.java [hunk 1]
```diff
@@ -16,18 +16,27 @@
...
+    return StringUtils.isBlank(str);
   }
 
+  /**
+   * Judge if the system is windows.
+   *
+   * @return true if the current system is windows.
+   */
   public static boolean isOSWindows() {
-    String osName = System.getProperty("os.name");
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply apollo-core/src/main/java/com/ctrip/framework/foundation/internals/Utils.java]: Yes, the implementation is the same, so this pr is not urgent. I just think it would be more elegant to use the existing utils, if we plan to upgrade commons-lang to commons-lang3.

As you said, it's unnecessary to extra dependency to apollo-core, I very agree with you, so I remove these changes.

File: apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java [hunk 2]
```diff
@@ -57,32 +52,11 @@ public void isBlankFalseGivenWhitespacePadded() {
...
-        System.setProperty("os.name", "Windows 10");
-        assertTrue(Utils.isOSWindows());
+        if (actualOsName.startsWith("Windows")) {
+            assertTrue(Utils.isOSWindows());
+        } else {
+            assertFalse(Utils.isOSWindows());
+        }
     }
 
-    @Test
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java:64]: Have removed these changes of this class. 
But suggest simplifying this test method like `commons-lang`, don't need to create a new test method for each system.
```java
    @Test
    public void test_IS_OS() {
        final String osName = System.getProperty("os.name");
        if (osName == null) {
            assertFalse(SystemUtils.IS_OS_WINDOWS);
            assertFalse(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_SOLARIS);
            assertFalse(SystemUtils.IS_OS_LINUX);
            assertFalse(SystemUtils.IS_OS_MAC_OSX);
        } else if (osName.startsWith("Windows")) {
            assertFalse(SystemUtils.IS_OS_UNIX);
            assertTrue(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.startsWith("Solaris")) {
            assertTrue(SystemUtils.IS_OS_SOLARIS);
            assertTrue(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.toLowerCase(Locale.ENGLISH).startsWith("linux")) {
            assertTrue(SystemUtils.IS_OS_LINUX);
            assertTrue(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.startsWith("Mac OS X")) {
            assertTrue(SystemUtils.IS_OS_MAC_OSX);
            assertTrue(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.startsWith("OS/2")) {
            assertTrue(SystemUtils.IS_OS_OS2);
            assertFalse(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.startsWith("SunOS")) {
            assertTrue(SystemUtils.IS_OS_SUN_OS);
            assertTrue(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else if (osName.startsWith("FreeBSD")) {
            assertTrue(SystemUtils.IS_OS_FREE_BSD);
            assertTrue(SystemUtils.IS_OS_UNIX);
            assertFalse(SystemUtils.IS_OS_WINDOWS);
        } else {
            System.out.println("Can't test IS_OS value: " + osName);
        }
    }
```

File: apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java [hunk 3]
```diff
@@ -83,6 +83,6 @@ public void isOSWindowsFalseGivenAllUppercaseWindows() {
     @Test
     public void isOSWindowsFalseGivenAllLowercaseWindows() {
         System.setProperty("os.name", "windows 10");
-        assertFalse(Utils.isOSWindows());
+            assertFalse(Utils.isOSWindows());
```
Comment (nobodyiam) [inline apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java:86]: I think we need to also revert this change back.

Comment (ruanwenjun) [reply apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java]: Done

**Review by nobodyiam — APPROVED** [review]
LGTM
